### PR TITLE
[10.0] Migrate mrp_production_properties

### PR DESCRIPTION
**Pull request by bimajatiwijaya**
- add not existing model on odoo 10 (mrp.property, mrp.property.group)
- add new test

### CODE REVIEW

File: mrp_production_properties/README.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+1. Add access user mrp who will manage property
+2. Activate developer mode
+3. Go to Setting > Users select user
+4. Check "Manage Properties of Product"
+5. Add access user sale to select property on sales
+6. Follow poin 3
+7. Check "Properties on lines"
+
+.. figure:: path/to/local/image.png
```
Comment (LoisRForgeFlow) [inline mrp_production_properties/README.rst]: remove this

File: mrp_production_properties/README.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+1. User sales make customer order and select product property on line.
+2. User sales confirm sales order.
+3. User manufacture check MO which have reference and same property with SO.
+4. Finish production.
+5. Do delivery order sales.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Comment (LoisRForgeFlow) [inline mrp_production_properties/README.rst]: fill `repo_id` and `branch`.

File: mrp_production_properties/README.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (LoisRForgeFlow) [inline mrp_production_properties/README.rst]: remove this

**Review by LoisRForgeFlow — COMMENTED** [review]
Checking README

**Comment by bimajatiwijaya** [comment]
Any advice for codecov?
 I already add unit test on module.

File: mrp_production_properties/README.rst [hunk 4]
```diff
@@ -46,11 +43,7 @@ To use this module, you need to:
 
 .. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
    :alt: Try me on Runbot
-   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
-
-.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
-.. branch is "8.0" for example
-
+   :target: https://runbot.odoo-community.org/runbot/OCA/manufacture/10.0
```
Comment (LoisRForgeFlow) [inline mrp_production_properties/README.rst]: It has to be <https://runbot.odoo-community.org/runbot/129/10.0>

**Comment by LoisRForgeFlow** [comment]
@bimajatiwijaya check the lines in red in the [test](https://codecov.io/gh/OCA/manufacture/compare/1cfeb27d4eb4b881acc9eb2746d13824f64a486a...468e3a5c351676b8579ea5d9680313f7bc98abd6/diff) and try to add tests for them.

File: mrp_production_properties/models/mrp.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            if company_id or self.env.context.get('company_id'):
+                domain += [('company_id', '=', company_id or
+                            self.env.context.get('company_id'))]
+
+            bom_ids = self.search(domain, order='sequence, product_id')
+            for bom in bom_ids.sorted(key=lambda p: (p.sequence,
+                                                     p.product_id)):
+                for ctx_p in property_ids:
+                    if ctx_p in [p.id for p in bom.property_ids]:
+                        return bom
```
Comment (StefanRijnhart) [inline mrp_production_properties/models/mrp.py]: You return a bom based on the first matching property. That is a functional change compared to Odoo 9.0. From https://github.com/odoo/odoo/blob/9.0/addons/mrp/mrp.py#L222:
> Search a BoM which has all properties specified, or if you can not find one, you could
> pass a BoM without any properties with the smallest sequence

Any reason for this?

File: mrp_production_properties/models/mrp.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                return False
+            if picking_type:
+                domain += ['|', ('picking_type_id', '=', picking_type.id),
+                           ('picking_type_id', '=', False)]
+            if company_id or self.env.context.get('company_id'):
+                domain += [('company_id', '=', company_id or
+                            self.env.context.get('company_id'))]
+
+            bom_ids = self.search(domain, order='sequence, product_id')
+            for bom in bom_ids.sorted(key=lambda p: (p.sequence,
```
Comment (StefanRijnhart) [inline mrp_production_properties/models/mrp.py]: boms are already sorted as per `order` argument in the previous search.

**Comment by StefanRijnhart** [comment]
Thank you for restoring this functionality from Odoo 9.0!

@lreficent is it mandatory to reach the original score? The module contains more code now, and I think covering the red lines from codecov is not very useful.

File: mrp_production_properties/models/mrp.py (hunk 6 above)
Comment (bimajatiwijaya) [reply mrp_production_properties/models/mrp.py]: I will update code. Thank for review.

**Comment by StefanRijnhart** [comment]
Thanks for the update! The Travis CI build now fails on a simple problem in the Lint check: mrp_production_properties/models/mrp.py:75:33: E126 continuation line over-indented for hanging indent

Can you fix that?

**Comment by StefanRijnhart** [comment]
@bima-pci did you ever file a contributor license agreement? I cannot find it in the OCA database and the OCA BOT has not yet turned up to clear your commits. See https://odoo-community.org/page/cla. Can you clarify the situation, even if you have not and do not intend to sign the CLA?

**Comment by StefanRijnhart** [comment]
I've reimplemented the mrp property from Odoo 9.0 here: https://github.com/OCA/manufacture/pull/291 because of CLA woes and other problems. Note that I did not revive Alex Comba's old module that propagated the properties to the BoM as that was not part of the native Odoo feature in 9.0.

**Comment by LoisRForgeFlow** [comment]
There is no answer by the author, closing. Please, reopen if you think this is still relevant.

**Pull request closed without merging**